xcm/xcm-builder/src/barriers.rs

```diff
@@ -39,6 +43,8 @@ impl ShouldExecute for TakeWeightCredit {
 
 /// Allows execution from `origin` if it is contained in `T` (i.e. `T::Contains(origin)`) taking payments into
 /// account.
+///
+/// Only allows for asset-based XCMs.
```

Comment:
```suggestion
/// Only allows for `TeleportAsset`, `WithdrawAsset` and `ReserveAssetDeposit` XCMs because they are the only ones
/// that place assets in holding to pay for execution.
```